app/views/alchemy/admin/resources/_form.html.erb

```diff
@@ -1,4 +1,4 @@
-<%= alchemy_form_for resource_instance_variable, url: resource_path(resource_instance_variable) do |f| %>
+<%= alchemy_form_for resource_instance_variable, url: resource_path(resource_instance_variable, q: params[:q]) do |f| %>
```

Comment:
It does work with the GET parameters; thanks to Rails. If I put it into a hidden field, I have to serialize and deserialize the object though (`to_json` in the view, `JSON.parse()` in the controller).

There's also the `delete` action that suffers from a similar thing: The delete URL gets converted to a POST, and there I don't have the option of adding hidden fields when using the `link_to` helper. 

In the end, I do vote for keeping the `GET` parameters, even though they're not so standard.